core/sql/optimizer/BindRelExpr.cpp

```diff
@@ -7582,6 +7582,24 @@ RelExpr *Scan::bindNode(BindWA *bindWA)
         }
     }
 
+   if (naTable->isHiveTable() && 
+       !(naTable->getClusteringIndex()->getHHDFSTableStats()->isOrcFile() ||
+	 naTable->getClusteringIndex()->getHHDFSTableStats()
+	 ->isSequenceFile()) &&
+       (CmpCommon::getDefaultNumeric(HDFS_IO_BUFFERSIZE_BYTES) == 0) && 
+       (naTable->getRecordLength() >
+	CmpCommon::getDefaultNumeric(HDFS_IO_BUFFERSIZE)*1024))
+     {
+       // do not raise error if buffersize is set though buffersize_bytes.
```

Comment:
Another thing you can do is to make HDFS_IO_BUFFERSIZE a FLOAT, then you can specify sizes of less than one KB as well. That would be less confusing. I'm not sure I understand how you wanted to use this as a backdoor and don't have a strong opinion, you will know better how you want to use it.